src/radiation/source_terms_multi_group.hpp

```diff
@@ -616,9 +616,10 @@ void RadSystem<problem_t>::AddSourceTermsMultiGroup(array_t &consVar, arrayconst
 
 		// load radiation energy source term
 		// plus advection source term (for well-balanced/SDC integrators)
+		// Note that radEnergySource should contain the luminosity volume density, L / V; unit: erg s^-1 cm^-3
 		quokka::valarray<double, nGroups_> Src;
 		for (int g = 0; g < nGroups_; ++g) {
-			Src[g] = dt * (chat * radEnergySource(i, j, k, g));
+			Src[g] = dt * (chat / c * radEnergySource(i, j, k, g));
```

Comment:
This seems like an unrelated change that should go in a separate PR.

Reply:
This is related. I found the original definition of the source term (which I adapted from your original code) confusing. There is a factor of `c` difference from my current implementation. I assume radEnergySource contain the luminosity density (erg s^-1 cm^-3), so the conversion from radEnergySource to Src should be `Src = dt * (chat/c) * radEnergySource`.